### PR TITLE
Add a "host" crate for autodetecting the host.

### PR DESCRIPTION
**Pull request by sunfishcode**
I'm open to suggestions here.

### CODE REVIEW

**Comment by stoklund** [comment]
Just a couple high-level comments:

1. I would prefer a more specific crate name than `cretonne-host`. How about `cretonne-cpuid`?
2. I really want to discourage tests that behave differently on different hosts. Specifically, the `lookup()` function provided here is a little too close to that kind of use by making it easy to write `isa host` in a test file somewhere. Would it make sense to have an API that is more focused on "I want to generate code for running on this host CPU", and that doesn't let you configure a cross compiler?

**Comment by sunfishcode** [comment]
`cretonne-cpuid` sounds somewhat x86-specific. How about `cretonne-native` to reflect that it's focused on native compilation and doesn't support cross compilation?

I've now updated the PR with patches that change the API to be less like `lookup` and that doesn't have a builtin fallback to cross compilation. Does this look right?

File: lib/native/.gitignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+target/
+**/*.rs.bk
+Cargo.lock
```
Comment (stoklund) [inline lib/native/.gitignore]: This should be covered by the top-level `.gitignore`, no?

File: lib/native/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+pub fn builders() -> Result<(settings::Builder, isa::Builder), ()> {
+    let mut flag_builder = settings::builder();
+
+    // TODO: Add RISC-V support once Rust supports it.
+
+    let name = if cfg!(target_arch = "x86") {
+        parse_x86_cpuid(&mut flag_builder);
+        "intel"
+    } else if cfg!(target_arch = "x86_64") {
+        flag_builder.enable("is_64bit").unwrap();
```
Comment (stoklund) [inline lib/native/src/lib.rs]: This probably simpler with `if cfg!(target_pointer_width = "64")`.

File: lib/native/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+            match err {
+                isa::LookupError::Unknown => panic!(),
+                isa::LookupError::Unsupported => (()),
+            }
+        },
+    )
+}
+
+#[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
+fn parse_x86_cpuid(flag_builder: &mut settings::Builder) {
```
Comment (stoklund) [inline lib/native/src/lib.rs]: The flags set by this function are Intel-specific, so they live in the `isa::Builder`. The `settings::Builder` deals with ISA-independent flags.

File: lib/native/.gitignore (hunk 1 above)
Comment (sunfishcode) [reply lib/native/.gitignore]: Ah, that was copied from `lib/wasm/.gitignore`, which is probably an leftover from when that code lived in its own repo. I'll remove both.

File: lib/native/src/lib.rs (hunk 2 above)
Comment (sunfishcode) [reply lib/native/src/lib.rs]: Fixed.

File: lib/native/src/lib.rs (hunk 3 above)
Comment (sunfishcode) [reply lib/native/src/lib.rs]: Fixed.

**Pull request merged**